updatehub/src/mem_drain.rs

```diff
@@ -26,11 +26,8 @@ struct LogRecord {
 }
 
 impl MemDrain {
-    pub fn clear(&self) {
-        self.records.lock().unwrap().clear();
-    }
-
     pub fn start_logging(&mut self) {
+        self.records.lock().unwrap().clear();
```

Comment:
I am wondering if we couldn't use this more directly and avoid the inner calls to the buffer and lock on the inner code.

Another minor thing is the commit log. It could get some love and describe what this is intending to do.

Reply:
We could avoid the mutex for the records field using a [RefCell for interior mutability](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html) but we can't access the field freely as we need to mutate it on log call that is `&self` and not `&mut self`. And if we did that we would have to find a way to handle the case where a mutable reference to the field is not available because some other log call maybe running, I'm not sure if this does actually happen, but the type signature doesn't ensure that for us.

Reply:
Ok, I'll merge this for now but please open an internal task so we can review this in the future and try to improve this.